iphone/cli/hooks/frameworks.js

```diff
@@ -728,28 +749,51 @@ class FrameworkInspector {
 	 * Inspects the framework under the given path and returns a new {@link FrameworkInfo}
 	 * instance for it.
 	 *
-	 * @param {String} frameworkPath - Path to the framwork to inspect
-	 * @return {Promise}
+	 * @param {String} frameworkPath - Path to the framework to inspect
+	 * @return {Promise<FrameworkInfo>}
 	 * @access public
 	 */
-	inspect(frameworkPath) {
-		let frameworkMatch = frameworkPath.match(frameworkPattern);
-		let frameworkName = frameworkMatch[1];
-		let binaryPathAndFilename = path.join(frameworkPath, frameworkName);
-		return this.detectBinaryTypeAndArchitectures(binaryPathAndFilename).then((result) => {
-			let frameworkInfo = new FrameworkInfo(frameworkName, frameworkPath, result.type, result.architectures);
-			let archs = Array.from(frameworkInfo.architectures.values()).join(', ');
-			this._logger.debug(`Found framework ${frameworkName.green} (type: ${result.type}, archs: ${archs}) at ${frameworkPath.cyan}`);
-			return frameworkInfo;
-		});
+	async inspect(frameworkPath) {
+		const packageExtension = path.extname(frameworkPath);
+		const frameworkName = path.basename(frameworkPath, packageExtension);
+		let frameworkInfo;
+		let type;
+		const meta = { name: frameworkName, path: frameworkPath };
+		if (packageExtension === '.framework') {
+			const binaryPathAndFilename = path.join(frameworkPath, frameworkName);
+			const result = await this.detectBinaryTypeAndArchitectures(binaryPathAndFilename);
+			meta.type = type = result.type;
+			meta.architectures = result.architectures;
+			frameworkInfo = new FrameworkInfo(meta);
+		} else {
+			const archs = new Set();
+			const supportedPlatforms = new Set();
+			const xcFrameworkInfo = await parsePlist(path.join(frameworkPath, 'Info.plist'));
+			for (const libInfo of xcFrameworkInfo.AvailableLibraries) {
+				const libraryPath = path.join(frameworkPath, libInfo.LibraryPath);
+				if (!type) {
+					const result = await this.detectBinaryTypeAndArchitectures(libraryPath);
+					type = result.type;
+				}
+				libInfo.SupportedArchitectures.forEach(a => archs.add(a));
+				supportedPlatforms.add(libInfo.SupportedPlatform);
```

Comment:
'SupportedPlatform' is always 'ios'. I think you want to use 'SupportedPlatformVariant' .

Reply:
I'm not sure i follow here. A XCFramework package can contain binaries for MacOS as well. I read the `SupportedPlatform` here to set the appropriate `PlatformFilter` in the Xcode build file entry. Why should `SupportedPlatform` always be `ios`?

Reply:
@janvennemann Vijay is referring to the info.plist value for `SupportedPlatform` always `=== 'ios'`. So you can't use that value to determine the filter later. You need to look at the optional `SupportedPlatformVariant` key's value, which `=== 'maccatalyst'` for macOS, and `=== 'simulator'` for the sim binaries.

Reply:
Ah yeah, i used a test framework to check possible values of the Info.plist and forgot to build the MacOS binary with the `-destination 'platform=macOS,variant=Mac Catalyst'` flag. Now that i rebuilt if with this flag i sea what you mean.

Reply:
Yes. @sgtcoolguy  is right. I was referring Info.plist's contents.